src/test/java/lambda/part3/exercise1/Exercise1.java

```diff
@@ -0,0 +1,86 @@
+package lambda.part3.exercise1;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@SuppressWarnings({"WeakerAccess", "unused"})
+public class Exercise1 {
+
+    @Test
+    public void mapEmployeesToLengthOfTheirFullNames() {
+        List<Employee> employees = getEmployees();
+        List<Integer> lengths = new ArrayList<>();
+
+        // функция извлечения информации о человеке из объекта сотрудника personExtractor: Employee -> Person
+        Function<Employee, Person> employeeToPersonFunc = Employee::getPerson;
+        // функция извлечения полного имени из информации о человеке fullNameExtractor: Person -> String
+        Function<Person, String> pesonToFullName = person -> person.getFirstName() + " " + person.getLastName();
+        // функция извлечения длины из строки stringLengthExtractor: String -> Integer
+        Function<String, Integer> fullNameToLength = String::length;
+        // функция извлечения длины полного имени из сотрудника fullNameLengthExtractor: Employee -> Integer
+
+        Function<Employee, Integer> fullNameLengthExtractor = employee ->
+                fullNameToLength.apply(pesonToFullName.apply(employeeToPersonFunc.apply(employee)));
+
+        Function<Employee, Integer> fullNameLengthExtractor2 = employee ->
+                employeeToPersonFunc.andThen(pesonToFullName.andThen(fullNameToLength)).apply(employee);
```

Comment:
better use `employeeToPersonFunc.andThen(pesonToFullName).andThen(fullNameToLength)` without `employee ->` and without `.apply(employee)`

Reply:
Done